common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java

```diff
@@ -287,6 +287,20 @@ public String decrypt(final String encryptedBlob) throws GeneralSecurityExceptio
             return encryptedBlob;
         }
 
+        // Try to read keystore key - to verify how often this is invoked before the migration is done.
+        // TODO: remove this whole try-catch clause once the experiment is done.
+        if (mTelemetryCallback != null) {
+            try {
+                final SecretKey key = loadSecretKey(KeyType.KEYSTORE_ENCRYPTED_KEY);
+                if (key == null) {
+                    mTelemetryCallback.logEvent(mContext, methodName, false, "KEY_DECRYPTION_KEYSTORE_KEY_NOT_INITIALIZED");
+                }
+            } catch (Exception e) {
+                // Best effort.
```

Comment:
In addition to telemetry, should we log here?

Reply:
I think we don't need to, as this won't make much sense in log file.